tailcfg/tailcfg.go

```diff
@@ -2455,6 +2455,25 @@ type HealthChangeRequest struct {
 	NodeKey key.NodePublic
 }
 
+// SetDeviceAttributesRequest is a request to update the
+// current node's attributes.
+//
+// As of 2024-12-30, this is an experimental dev feature
+// for internal testing. See tailscale/corp#24690.
+type SetDeviceAttributesRequest struct {
+	Version CapabilityVersion
+	NodeKey key.NodePublic
+
+	// Update is a map of attributes to update.
+	// Attributes not in the map are left unchanged.
+	Update AttrUpdate
+}
+
+// AttrUpdate is a map of attributes to update.
+// Attributes not in the map are left unchanged.
```

Comment:
I'd suggest linking to https://tailscale.com/s/api-device-posture-attrs (which I am adding in https://github.com/tailscale/tailscale-www/pull/7256) to provide more context on supported keys and values.

Reply:
done